ext/pcntl/pcntl.c

```diff
@@ -965,7 +969,7 @@ PHP_FUNCTION(pcntl_signal)
 		/* since calling malloc() from within a signal handler is not portable,
 		 * pre-allocate a few records for recording signals */
 		int i;
-		for (i = 0; i < 32; i++) {
+		for (i = 0; i < MAX(NSIG-1, SIGRTMAX); i++) {
```

Comment:
This condition should be `i < MAX(NSIG, SIGRTMAX+1)`, otherwise SIGRTMAX cannot be set.